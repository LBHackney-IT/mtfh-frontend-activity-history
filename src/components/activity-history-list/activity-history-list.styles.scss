.mtfh-activity-history {
  b {
    font-weight: 600;
  }

  p {
    margin: 0;
  }
}

.lbh-simple-pagination {
  border-top: none;
}
