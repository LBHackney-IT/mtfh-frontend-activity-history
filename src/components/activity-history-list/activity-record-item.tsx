import React from 'react';

import { Tr, Th, Td } from '@mtfh/common';

export interface ActivityRecordItemProps {
    date: any;
    category: any;
    editDetails: any;
    editedBy: any;
}

export const ActivityRecordItem = ({
    date,
    category,
    editDetails,
    editedBy,
}: ActivityRecordItemProps): JSX.Element | null => {
    return (
        <Tr className="govuk-table__row mtfh-activity-history">
            <Td>{date}</Td>
            <Td>{category}</Td>
            <Td>{editDetails}</Td>
            <Th>{editedBy}</Th>
        </Tr>
    );
};
