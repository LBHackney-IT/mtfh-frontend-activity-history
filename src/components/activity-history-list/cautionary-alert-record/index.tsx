export * from "./cautionarty-alert-record";
