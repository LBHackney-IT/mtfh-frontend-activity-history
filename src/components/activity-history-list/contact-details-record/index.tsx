export * from './contact-details-record';
