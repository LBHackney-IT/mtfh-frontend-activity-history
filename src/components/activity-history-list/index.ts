export * from './activity-history-list';
