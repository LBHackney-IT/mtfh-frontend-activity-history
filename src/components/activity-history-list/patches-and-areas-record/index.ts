export * from "./patches-and-areas-record";
