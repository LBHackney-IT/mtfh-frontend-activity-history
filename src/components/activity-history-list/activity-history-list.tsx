import React, { useMemo } from "react";

import { ContactDetailsActivityRecord } from "./contact-details-record";
import { PersonEqualityInformationActivityRecord } from "./person-equality-information-record";
import { PersonActivityRecord } from "./person-record";
import { TenurePersonActivityRecord } from "./tenure-person-record";
import { TenureActivityRecord } from "./tenure-record";

import { useReferenceData } from "@mtfh/common/lib/api/reference-data/v1";
import {
  Center,
  ErrorSummary,
  SimplePagination,
  SimplePaginationButton,
  Spinner,
  Table,
  Tbody,
  Th,
  Thead,
  Tr,
} from "@mtfh/common/lib/components";

import { locale, useActivityHistory } from "@services";

import "./activity-history-list.styles.scss";

const { tableDate, tableCategory, tableEditDetails, tableEdittedBy, noActivityHistory } =
  locale.activities;

function NoActivityHistory() {
  return <p className="lbh-label">{noActivityHistory}</p>;
}
export interface ActivityHistoryListProps {
  targetId: string;
}

export const ActivityHistoryList = ({
  targetId,
}: ActivityHistoryListProps): JSX.Element => {
  const { data, size, setSize, error } = useActivityHistory(targetId);

  const response = useMemo(() => {
    if (!data) return null;
    return data[size - 1];
  }, [data, size]);

  const { data: referenceData, error: errorReferenceData } = useReferenceData<
    | "answers"
    | "religion-belief"
    | "age-bracket"
    | "ethnic-group-a"
    | "gender"
    | "sexual-orientation"
  >({
    category: "equality-information",
  });

  if (errorReferenceData) {
    return (
      <ErrorSummary
        id="equality-information-error"
        title={locale.errors.unexpectedResponse}
        description={locale.errors.unexpectedResponseDescription}
      />
    );
  }

  if (error?.response?.status === 404) {
    return <NoActivityHistory />;
  }

  if (!response || !referenceData) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const {
    results: activityHistory,
    paginationDetails: { nextToken },
  } = response;

  return (
    <div>
      <Table>
        <Thead>
          <Tr>
            <Th>{tableDate}</Th>
            <Th>{tableCategory}</Th>
            <Th>{tableEditDetails}</Th>
            <Th>{tableEdittedBy}</Th>
          </Tr>
        </Thead>
        <Tbody>
          {activityHistory.map((activity, index) => {
            const { targetType } = activity;
            if (targetType === "person") {
              return <PersonActivityRecord key={index} personRecord={activity} />;
            }
            if (targetType === "contactDetails") {
              return (
                <ContactDetailsActivityRecord
                  key={index}
                  contactDetailsRecord={activity}
                />
              );
            }
            if (targetType === "tenure") {
              return <TenureActivityRecord key={index} tenureRecord={activity} />;
            }
            if (targetType === "tenurePerson") {
              return (
                <TenurePersonActivityRecord key={index} tenurePersonRecord={activity} />
              );
            }
            if (targetType === "personEqualityInformation") {
              return (
                <PersonEqualityInformationActivityRecord
                  referenceData={referenceData}
                  key={index}
                  personEqualityInformationRecord={activity}
                />
              );
            }
            return null;
          })}
        </Tbody>
      </Table>
      <SimplePagination>
        {size !== 1 && (
          <SimplePaginationButton variant="previous" onClick={() => setSize(size - 1)}>
            Previous
          </SimplePaginationButton>
        )}
        {nextToken && (
          <SimplePaginationButton variant="next" onClick={() => setSize(size + 1)}>
            Next
          </SimplePaginationButton>
        )}
      </SimplePagination>
    </div>
  );
};
