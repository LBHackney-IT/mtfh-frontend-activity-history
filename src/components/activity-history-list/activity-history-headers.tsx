import React from "react";

import { Th, Thead, Tr } from "@mtfh/common/lib/components";

import { EntityType, locale } from "@services";

const {
  tableBy,
  tableDate,
  tableCaseDetails,
  tableCategory,
  tableEditDetails,
  tableEdittedBy,
} = locale.activities;

export const ActivityHistoryHeaders = ({
  entityType,
}: {
  entityType: EntityType;
}): JSX.Element => {
  let headers = (
    <Tr>
      <Th>{tableDate}</Th>
      <Th>{tableCategory}</Th>
      <Th>{tableEditDetails}</Th>
      <Th>{tableEdittedBy}</Th>
    </Tr>
  );
  if (entityType === "process") {
    headers = (
      <Tr>
        <Th>{tableDate.charAt(0).toUpperCase() + tableDate.slice(1)}</Th>
        <Th>{tableCategory.charAt(0).toUpperCase() + tableCategory.slice(1)}</Th>
        <Th>{tableCaseDetails}</Th>
        <Th>{tableBy}</Th>
      </Tr>
    );
  }
  return <Thead>{headers}</Thead>;
};
