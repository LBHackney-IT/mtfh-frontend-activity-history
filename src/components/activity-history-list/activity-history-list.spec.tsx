import React from "react";

import { screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import {
  mockAddedPersonToTenure,
  mockCreatedAddressWithContactTypeAsString,
  mockCreatedEmail,
  mockCreatedEmailWithContactTypeAsString,
  mockCreatedPerson,
  mockCreatedPersonEqualityInformation,
  mockCreatedPhoneNumber,
  mockCreatedPhoneNumberWithContactTypeAsString,
  mockCreatedTenure,
  mockEdittedTenureWithInValidParam,
  mockMigratedPerson,
  mockMigratedPersonEqualityInformation,
  mockMigratedTenure,
  mockRemovedEmail,
  mockRemovedPersonFromTenure,
  mockRemovedPhoneNumber,
  mockUpdatedDateOfBirth,
  mockUpdatedFirstName,
  mockUpdatedIdentifications,
  mockUpdatedLanguages,
  mockUpdatedPersonEqualityInformation,
  mockUpdatedPlaceOfBirth,
  mockUpdatedTenure,
} from "../../mocks";
import { get, routeRender } from "../../test-utils";
import { ActivityHistoryList } from "./activity-history-list";

import { locale } from "@services";

beforeEach(() => {
  const mockEqualityData = {
    "age-bracket": [
      { code: "eightyFiveandPlus", value: "85+" },
      { code: "underSixteen", value: "Under 16" },
    ],
    answers: [
      { code: "no", value: "No" },
      { code: "yes", value: "Yes" },
      { code: "preferNotToSay", value: "Prefer not to say" },
    ],
    "ethnic-group-a": [
      { code: "mixedBackground", value: "Mixed background" },
      { code: "other", value: "Other - please describe" },
      { code: "whiteOrWhiteBritish", value: "White or White British" },
    ],
    gender: [
      { code: "f", value: "Female" },
      { code: "m", value: "Male" },
      { code: "o", value: "Other" },
    ],
    "religion-belief": [
      { code: "other", value: "Other" },
      { code: "secularBeliefs", value: "Secular beliefs" },
      { code: "sikh", value: "Sikh" },
    ],
    "sexual-orientation": [
      { code: "bisexual", value: "Bisexual" },
      { code: "heterosexual", value: "Heterosexual" },
      { code: "gayMan", value: "Gay man" },
      { code: "lesbianOrGayWoman", value: "Lesbian or Gay woman" },
      { code: "other", value: "Other" },
    ],
  };
  get("/api/v1/reference-data", mockEqualityData);
});

test("it renders no comments with no results", async () => {
  get("/api/activityhistory", {}, 404);
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/No activity history/);
});

test.skip("it pages the results upon clicking next and previous", async () => {
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Next/);

  get("/api/activityhistory", {
    results: [mockMigratedPerson],
    paginationDetails: {
      nextToken: null,
    },
  });
  userEvent.click(screen.getByText(/Next/));

  await screen.findByText(/Person migrated/);

  userEvent.click(screen.getByText(/Previous/));

  await screen.findByText(/First name/);
});

test("it renders correctly", () => {
  routeRender(<ActivityHistoryList targetId="123" />);
  expect(screen.getByText(/Loading/)).toBeInTheDocument();
});

test("it displays a Person created on the activity history list for a new person record", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedPerson],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Person created/);
});

test("it displays a Person editted on the activity history list for a new person record", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedFirstName],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Edit to person/);
});

test("it pages the results", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedPerson, mockUpdatedFirstName],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/date/);
  await screen.findByText(/category/);
  await screen.findByText(/edit details/);
  await screen.findByText(/edited by/);

  await screen.findByText(/Person created/);
});

test("it pages the results for migrated person information", async () => {
  get("/api/activityhistory", {
    results: [mockMigratedPerson],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Person migrated/);
});

test("it does not render pagination unnecessarily", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedFirstName],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => expect(screen.queryByText(/Next/)).toBe(null));
});

test("it should display change in Languages", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedLanguages],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Languages/);
  await waitFor(() => expect(container).toMatchSnapshot());
});

test("it should display change in Identifications", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedIdentifications],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Identitifications/);
  await waitFor(() => expect(container).toMatchSnapshot());
});

test("it should display a row for created phone number (contactType is number)", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedPhoneNumber],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => expect(container).toMatchSnapshot());
  await screen.findByText(/Added/);
  await screen.findByText(/07123123123/);
});

test("it should display a row for created email (contactType is number)", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedEmail],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => expect(container).toMatchSnapshot());
  await screen.findByText(/Added/);
  await screen.findByText(/email@address.com/);
});

test("it should display a row for removed phone number (contactType is number)", async () => {
  get("/api/activityhistory", {
    results: [mockRemovedPhoneNumber],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => expect(container).toMatchSnapshot());

  await screen.findByText(/Removed/);
  await screen.findByText(/07123123123/);
});

test("it should display a row for removed email (contactType is number)", async () => {
  get("/api/activityhistory", {
    results: [mockRemovedEmail],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => expect(container).toMatchSnapshot());

  await screen.findByText(/Removed/);
  await screen.findByText(/email@address.com/);
});

test("it should display a row for change in date of birth", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedDateOfBirth],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => expect(container).toMatchSnapshot());

  await screen.findByText(/Date of birth/);
  await screen.findByText("23/04/62");
});

test("it should display a row for change in place of birth", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedPlaceOfBirth],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Place of birth/);
  await screen.findByText(/London/);
});

test("it should display a row for created phone number (contactType is string)", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedPhoneNumberWithContactTypeAsString],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Added/);

  await screen.findByText(/Phone/);

  await screen.findByText(/07123123123/);
});

test("it should display a row for created email (contactType is string)", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedEmailWithContactTypeAsString],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Added/);

  await screen.findByText(/Email/);
  await screen.findByText(/email@address.com/);
});

test("it should display a row for created address (contactType is string)", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedAddressWithContactTypeAsString],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Added/);

  await screen.findByText(/Address/);
  await screen.findByText(/An address with postcode/);
});

test("it should display a row for migrated tenure", async () => {
  get("/api/activityhistory", {
    results: [mockMigratedTenure],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Tenure migrated/);
});

test("it should display a row with parameter name if the param is not part of the updatable entity", async () => {
  get("/api/activityhistory", {
    results: [mockEdittedTenureWithInValidParam],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/invalidParam/);
});

test("it should display a row for created tenure", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedTenure],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => expect(screen.queryAllByText(/Tenure created/).length).toBe(2));
});

test("it should display a row for updated tenure status (Activity) details", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedTenure],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Active/);

  await screen.findByText(/Inactive/);
});

test("it should display a row for an added person to tenure", async () => {
  get("/api/activityhistory", {
    results: [mockAddedPersonToTenure],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => {
    expect(screen.getByText(locale.activities.personAddedToTenure)).toBeInTheDocument();
    expect(
      screen.getByText(mockAddedPersonToTenure.newData?.householdMembers[0].fullName),
    ).toBeInTheDocument();
    expect(screen.getByText("01/01/01")).toBeInTheDocument();
    expect(screen.getByText("Household member")).toBeInTheDocument();
  });
});

test("it should display a row for a removed person from tenure", async () => {
  get("/api/activityhistory", {
    results: [mockRemovedPersonFromTenure],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await waitFor(() => {
    expect(
      screen.getByText(locale.activities.personRemovedFromTenure),
    ).toBeInTheDocument();
    expect(
      screen.getByText(mockRemovedPersonFromTenure.oldData?.householdMembers[0].fullName),
    ).toBeInTheDocument();
    expect(screen.getByText("01/01/01")).toBeInTheDocument();
    expect(screen.getByText("Household member")).toBeInTheDocument();
  });
});

test("it should display a row for migrated equality information", async () => {
  get("/api/activityhistory", {
    results: [mockMigratedPersonEqualityInformation],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Equality information migrated/);
});

test("it should display a row for created equality information", async () => {
  get("/api/activityhistory", {
    results: [mockCreatedPersonEqualityInformation],
    paginationDetails: {
      nextToken: null,
    },
  });
  routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText(/Equality information created/);
});

test("it should display a row for updated equality information", async () => {
  get("/api/activityhistory", {
    results: [mockUpdatedPersonEqualityInformation],
    paginationDetails: {
      nextToken: null,
    },
  });
  const [{ container }] = routeRender(<ActivityHistoryList targetId="123" />);

  await screen.findByText("Age Group");
  await screen.findByText("85+");
  await screen.findByText("Under 16");

  await screen.findByText("Gender");
  await screen.findByText("Male (gender different to birth sex: No)");
  await screen.findByText("Another gender (gender different to birth sex: Yes)");

  await screen.findByText("Ethnicity");
  await screen.findByText("Mixed background");
  await screen.findByText("Another ethnic group");

  await screen.findByText("Religion or Belief");
  await screen.findByText("Secular beliefs");
  await screen.findByText("Another belief");

  await screen.findByText("Pregnancy or Maternity");
  await screen.findByText("01/05/21");
  await screen.findByText("[No entry]");

  await screen.findByText("Sexual Orientation");
  await screen.findByText("Heterosexual");
  await screen.findByText("Another orientation");

  await screen.findByText("Disabled");
  await screen.findByText("Caring Responsibilities");

  expect(container).toMatchSnapshot();
});
