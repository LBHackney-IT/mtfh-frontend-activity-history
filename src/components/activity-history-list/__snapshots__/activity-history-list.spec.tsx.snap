// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should display a row for created email 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
  >
    <svg
      class="mtfh-icon"
      focusable="false"
      height="50"
      stroke="#00703c"
      viewBox="0 0 42 42"
      width="50"
    >
      <title>
        Loading...
      </title>
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="5"
          transform="translate(3 3)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
            transform="rotate(112.708 18 18)"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`it should display a row for created phone number 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
  >
    <svg
      class="mtfh-icon"
      focusable="false"
      height="50"
      stroke="#00703c"
      viewBox="0 0 42 42"
      width="50"
    >
      <title>
        Loading...
      </title>
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="5"
          transform="translate(3 3)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
            transform="rotate(112.708 18 18)"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`it should display a row for removed email 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
  >
    <svg
      class="mtfh-icon"
      focusable="false"
      height="50"
      stroke="#00703c"
      viewBox="0 0 42 42"
      width="50"
    >
      <title>
        Loading...
      </title>
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="5"
          transform="translate(3 3)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
            transform="rotate(112.708 18 18)"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`it should display a row for removed phone number 1`] = `
<div>
  <div
    class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
  >
    <svg
      class="mtfh-icon"
      focusable="false"
      height="50"
      stroke="#00703c"
      viewBox="0 0 42 42"
      width="50"
    >
      <title>
        Loading...
      </title>
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="5"
          transform="translate(3 3)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
            transform="rotate(112.708 18 18)"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`it should display change in Identifications 1`] = `
<div>
  <div>
    <table
      class="govuk-table lbh-table"
    >
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__cell govuk-table__header"
          >
            date
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            category
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            edit details
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            edited by
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row govuk-table__row mtfh-activity-history"
        >
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                19/09/19
              </p>
              <p>
                03:12
              </p>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          >
            Edit to person
          </td>
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                <b>
                  Identitifications
                </b>
              </p>
              <p>
                Previously:
                 
                <b>
                  Passport, GB03654488992 seen
                </b>
              </p>
              <p>
                Changed to:
                 
                <b>
                  Passport, GB03654488992 seen
                </b>
              </p>
            </div>
          </td>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            Paul Fox
          </th>
        </tr>
      </tbody>
    </table>
    <nav
      class="lbh-simple-pagination"
    />
  </div>
</div>
`;

exports[`it should display change in Languages 1`] = `
<div>
  <div>
    <table
      class="govuk-table lbh-table"
    >
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__cell govuk-table__header"
          >
            date
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            category
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            edit details
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            edited by
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row govuk-table__row mtfh-activity-history"
        >
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                19/09/19
              </p>
              <p>
                03:12
              </p>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          >
            Edit to person
          </td>
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                <b>
                  Languages
                </b>
              </p>
              <p>
                Previously:
                 
                <b>
                  English (primary)
                </b>
              </p>
              <p>
                Changed to:
                 
                <b>
                  English (primary), Abkhaz
                </b>
              </p>
            </div>
          </td>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            Paul Fox
          </th>
        </tr>
      </tbody>
    </table>
    <nav
      class="lbh-simple-pagination"
    />
  </div>
</div>
`;
